Dockerfile

```diff
@@ -0,0 +1,232 @@
+RUN cd /tmp/ \
+    && git clone --recursive https://github.com/broadgsa/gatk-protected.git \
+    && cd /tmp/gatk-protected && git checkout tags/${gatk_version} \
+    && sed -i 's/^import oracle.jrockit.jfr.StringConstantPool;/\/\/import oracle.jrockit.jfr.StringConstantPool;/' ./public/gatk-tools-public/src/main/java/org/broadinstitute/gatk/tools/walkers/varianteval/VariantEval.java \
+    && mv /tmp/gatk-protected /opt/${gatk_dir_name}-${gatk_version}
+RUN cd /opt/ && unzip /tmp/${maven_package_name}-bin.zip \
+    && rm -rf /tmp/${maven_package_name}-bin.zip LICENSE NOTICE README.txt \
+    && cd /opt/ \
+    && cd /opt/${gatk_dir_name}-${gatk_version} && /opt/${maven_package_name}/bin/mvn verify -P\!queue \
+    && mv /opt/${gatk_dir_name}-${gatk_version}/protected/gatk-package-distribution/target/gatk-package-distribution-${gatk_version}.jar /opt/GenomeAnalysisTK.jar \
+    && rm -rf /opt/${gatk_dir_name}-${gatk_version} /opt/${maven_package_name}
+
+###############
+#Strelka 2.7.1#
+###############
+RUN apt-get update && apt-get install -y \
+    bzip2 \
+    g++ \
+    make \
+    perl-doc \
+    python \
+    rsync \
+    wget \
+    zlib1g-dev
+
+ENV STRELKA_INSTALL_DIR /opt/strelka/
+
+RUN wget https://github.com/Illumina/strelka/releases/download/v2.7.1/strelka-2.7.1.centos5_x86_64.tar.bz2 \
+    && tar xf strelka-2.7.1.centos5_x86_64.tar.bz2 \
+    && rm -f strelka-2.7.1.centos5_x86_64.tar.bz2 \
+    && mv strelka-2.7.1.centos5_x86_64 $STRELKA_INSTALL_DIR
+
+#strelka requires a couple steps to run, so add a helper script to sequence those
+COPY strelka_helper.pl /usr/bin/strelka_helper.pl
+
+###############
+#Varscan 2.4.2#
+###############
+RUN apt-get update && apt-get install -y \
+    default-jre \
+    wget
+
+ENV VARSCAN_INSTALL_DIR=/opt/varscan
+
+WORKDIR $VARSCAN_INSTALL_DIR
+RUN wget https://github.com/dkoboldt/varscan/releases/download/2.4.2/VarScan.v2.4.2.jar && \
+    ln -s VarScan.v2.4.2.jar VarScan.jar
+
+COPY intervals_to_bed.pl /usr/bin/intervals_to_bed.pl
+
+##############
+#HTSlib 1.3.2#
+##############
+ENV HTSLIB_INSTALL_DIR=/opt/samtools
+
+WORKDIR /tmp
+RUN wget https://github.com/samtools/htslib/releases/download/1.3.2/htslib-1.3.2.tar.bz2 && \
+    tar --bzip2 -xvf htslib-1.3.2.tar.bz2
+
+WORKDIR /tmp/htslib-1.3.2
+RUN ./configure  --enable-plugins --prefix=$HTSLIB_INSTALL_DIR/software/htslib-1.3.2 && \
+    make && \
+    make install
+
+################
+#Samtools 1.3.1#
+################
+RUN apt-get update && apt-get install -y \
+    bzip2 \
+    g++ \
+    make \
+    ncurses-dev \
+    wget \
+    zlib1g-dev
+
+ENV SAMTOOLS_INSTALL_DIR=/opt/samtools
+
+WORKDIR /tmp
+RUN wget https://github.com/samtools/samtools/releases/download/1.3.1/samtools-1.3.1.tar.bz2 && \
+    tar --bzip2 -xf samtools-1.3.1.tar.bz2
+
+WORKDIR /tmp/samtools-1.3.1
+RUN ./configure --with-htslib=$SAMTOOLS_INSTALL_DIR/software/htslib-1.3.2 --prefix=$SAMTOOLS_INSTALL_DIR && \
```

Comment:
If the above change is made, this would be `--with-htslib=$SAMTOOLS_INSTALL_DIR/htslib-1.3.2`